ConwaysGameOfLife.Tests/UniverseShould.cs

```diff
@@ -154,9 +154,113 @@ public void WhenTwoUniversesInstantiated_NotBeEqual()
+                new Location(0, 0), // x + 1, y + 1
+                new Location(0, 1),
+                new Location(0, 2),
+                new Location(1, 0),
+                new Location(1, 2),
+                new Location(2, 0),
+                new Location(2, 1),
+                new Location(2, 2)
+            };
             
-            Assert.Equal("Object reference not set to an instance of an object.", result.Message);
+            var result = universe.GetCellNeighbourLocations(cell);
+
+            Assert.True(UniverseHelper.ListsOfLocationsAreEqual(expected, result));
+        }
+
+        [Fact]
+        public void WhenGivenACellOnBoundaryX_ReturnCorrectNeighbourLocationsListInOrder()
+        {
+            var universe = new Universe(3, 3);
+            var cell = new Cell(CellState.Dead, 0, 1);
+            var expected = new List<Location>() {
+                new Location(2, 0),
+                new Location(2, 1),
+                new Location(2, 2),
+                new Location(0, 0),
+                new Location(0, 2),
+                new Location(1, 0),
+                new Location(1, 1),
+                new Location(1, 2)
+            };
+
+            var result = universe.GetCellNeighbourLocations(cell);
+
+            Assert.True(UniverseHelper.ListsOfLocationsAreEqual(expected, result));
+        }
+
+        [Fact]
+        public void WhenGivenABoundaryCellX_ReturnCorrectNeighbourLocationsListOutOfOrder()
+        {
+            var universe = new Universe(3, 3);
+            var cell = new Cell(CellState.Dead, 0, 1);
+            var expected = new List<Location>() {
+                new Location(2, 1),
+                new Location(2, 0),
+                new Location(2, 2),
+                new Location(0, 0),
+                new Location(0, 2),
+                new Location(1, 0),
+                new Location(1, 1),
+                new Location(1, 2)
+            };
+
+            var result = universe.GetCellNeighbourLocations(cell);
+
+            Assert.True(UniverseHelper.ListsOfLocationsAreEqual(expected, result));
+        }
+
+        [Fact]
+        public void WhenGivenABoundaryCellY_ReturnCorrectNeighbourLocationsListOutOfOrder()
+        {
+            var universe = new Universe(3, 3);
+            var cell = new Cell(CellState.Dead, 1, 0);
+            var expected = new List<Location>() {
+                new Location(2, 1),
+                new Location(2, 0),
+                new Location(2, 2),
+                new Location(0, 0),
+                new Location(0, 2),
+                new Location(0, 1),
+                new Location(1, 1),
+                new Location(1, 2)
+            };
+
+            var result = universe.GetCellNeighbourLocations(cell);
+
+            Assert.True(UniverseHelper.ListsOfLocationsAreEqual(expected, result));
+        }
+
+        [Fact]
+        public void WhenGivenABoundaryCornerCell_ReturnCorrectNeighbourLocationsListOutOfOrder()

```

Comment:
love that you're testing several "kinds" of locations i.e.
- non boundary
- single boundary
- dual boundary

could think about extending this with `[Theory]` to become:
- non boundary
- single boundary on y axis (case each for top and bottom)
- single boundary on x axis (case each for left and right)
- all four corners?